database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -204,6 +256,36 @@ public boolean removeRecordFromTable(Class<? extends BaseEntity> entityClass, In
         }
     }
 
+    @Override
+    public boolean removeSpecificRecords(Class<? extends BaseEntity> entityClass, List<Integer> ids) {
+        if (ids == null || ids.isEmpty()) {
+            throw new IllegalArgumentException("IDs list cannot be null or empty");
+        }
+
+        String tableName = entityClass.getSimpleName();
+        String placeholders = ids.stream()
+                .map(id -> "?")
+                .collect(Collectors.joining(", "));
+        String deleteRecordSQL = "DELETE FROM " + tableName + " WHERE id IN (" + placeholders + ")";
+
+        LOG.info("Executing SQL: {}", deleteRecordSQL);
+
+        Connection connection = connectionPool.getConnection();
+
+        try (PreparedStatement preparedStatement = connection.prepareStatement(deleteRecordSQL)) {
+            for (int i = 0; i < ids.size(); i++) {
+                preparedStatement.setInt(i + 1, ids.get(i));
+            }
+            preparedStatement.executeUpdate();
```

Comment:
Please use `executeBatch(`) to insert all records in one operation. Check how to add records in batches, you can't just execute query with multiple `drop` operators.